### PR TITLE
Update foundry.md

### PR DESCRIPTION
**Pull request by wuzhong-papermoon**
Original PR：https://github.com/moonbeam-foundation/moonbeam-docs/pull/847

### CODE REVIEW

File: builders/build/eth-api/dev-env/foundry.md [hunk 1]
```diff
@@ -99,13 +99,17 @@ forge install OpenZeppelin/openzeppelin-contracts
 forge build
 ```
 
-![Foundry Contract Compile](/images/builders/build/eth-api/dev-env/foundry/foundry-1.webp)
+--8<-- 'code/builders/build/eth-api/dev-env/foundry/terminal/compile.md'
```
Comment (eshaben) [inline builders/build/eth-api/dev-env/foundry.md:102]: Closest I could get to comment but line 88 `Github repo` should be changed to `GitHub repository` (note the capital H in GitHub)

File: builders/build/eth-api/dev-env/foundry.md [hunk 2]
```diff
@@ -143,11 +147,45 @@ forge build
...
+forge script script/MyToken.s.sol --rpc-url https://rpc.api.moonbase.moonbeam.network --broadcast
+```
+
+如果您的脚本执行成功，您的终端应展示一下内容：
 
-![Foundry Contract Deploy](/images/builders/build/eth-api/dev-env/foundry/foundry-2.webp)
+--8<-- 'code/builders/build/eth-api/dev-env/foundry/terminal/script.md'
 
-恭喜您，您的合约已上线！保存地址，用于后续合约实例交互。
+这样就成了。有关使用Foundry进行Solidity脚本编写的更多信息，请查看[Foundry文档](https://book.getfoundry.sh/tutorials/solidity-scripting){target=_blank}。
```
Comment (eshaben) [inline builders/build/eth-api/dev-env/foundry.md:188]:
```suggestion
这样就成了。有关使用Foundry进行Solidity脚本编写的更多信息，请查看[Foundry文档](https://book.getfoundry.sh/tutorials/solidity-scripting){target=\_blank}。
```

Comment (eshaben) [reply builders/build/eth-api/dev-env/foundry.md:188]: Closest I could get, but on line 194, can you please change `cast` to `Cast` with a capital C

and also, please change all instances of `YOUR_CONTRACT_ADDRESS` to `INSERT_YOUR_CONTRACT_ADDRESS` to follow convention

File: builders/build/eth-api/dev-env/foundry.md [hunk 3]
```diff
@@ -143,11 +147,45 @@ forge build
...
+
+!!! 请记住
+    请记住永远不要像上面那样将生产私钥存储在文件中。 这个例子仅用于演示目的。
+
+请注意，即使上述脚本不会被部署，它仍需要符合典型Solidity合约的所有格式，例如pragma语句。
+
+您可以使用以下命令部署 `MyToken.sol` 合约。请记住它将按顺序执行所有相关步骤。对于此示例，Foundry将首先尝试本地模拟和利用提供的 RPC进行模拟，然后才部署合约。如果任何模拟失败，Foundry都不会继续部署。
+
+```bash
+forge script script/MyToken.s.sol --rpc-url https://rpc.api.moonbase.moonbeam.network --broadcast
```
Comment (eshaben) [inline builders/build/eth-api/dev-env/foundry.md:181]:
```suggestion
forge script script/MyToken.s.sol --rpc-url {{ networks.moonbase.rpc_url }} --broadcast
```

File: builders/build/eth-api/dev-env/foundry.md [hunk 4]
```diff
@@ -185,7 +223,7 @@ Foundry包括cast，一个用于执行以太坊RPC调用的CLI。
 0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000074d79546f6b656e00000000000000000000000000000000000000000000000000
 ```
 
-此数据非可读，但您可以使用cast将其转换成您想要的格式。在这种情况下，数据是文本，因此您可以将其转换为ascii字符以查看“My Token”：
+此数据非可读，但您可以使用cast将其转换成您想要的格式。在这种情况下，数据是文本，因此您可以将其转换为ASCII字符以查看“My Token”：
 
 ![Foundry Contract View](/images/builders/build/eth-api/dev-env/foundry/foundry-3.webp)
```
Comment (eshaben) [inline builders/build/eth-api/dev-env/foundry.md:228]: This image is removed and is replaced by:

--8<-- 'code/builders/build/eth-api/dev-env/foundry/terminal/cast.md'

File: builders/build/eth-api/dev-env/foundry.md [hunk 5]
```diff
@@ -185,7 +223,7 @@ Foundry包括cast，一个用于执行以太坊RPC调用的CLI。
 0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000074d79546f6b656e00000000000000000000000000000000000000000000000000
 ```
 
-此数据非可读，但您可以使用cast将其转换成您想要的格式。在这种情况下，数据是文本，因此您可以将其转换为ascii字符以查看“My Token”：
+此数据非可读，但您可以使用cast将其转换成您想要的格式。在这种情况下，数据是文本，因此您可以将其转换为ASCII字符以查看“My Token”：
```
Comment (eshaben) [inline builders/build/eth-api/dev-env/foundry.md:226]:
```suggestion
此数据非可读，但您可以使用Cast将其转换成您想要的格式。在这种情况下，数据是文本，因此您可以将其转换为ASCII字符以查看“My Token”：
```

File: builders/build/eth-api/dev-env/foundry.md [hunk 6]
```diff
@@ -387,7 +425,7 @@ abi.encode(100, true, "Develop on Moonbeam")
 2. 加载存储的状态
 
     ```bash
-    chisel load
+    chisel load 1
```
Comment (eshaben) [inline builders/build/eth-api/dev-env/foundry.md:428]: on line 437, the image should be removed and replaced with:

--8<-- 'code/builders/build/eth-api/dev-env/foundry/terminal/save-state.md'

**Review by eshaben — CHANGES_REQUESTED** [review]
Nice job! The original PR here mentions that once we've completed translations, we need to delete the foundry images from the `moonbeam-docs` repo - so can you please open a PR that deletes them?

**Comment by eshaben** [comment]
@wuzhong-papermoon more changes were made to the Foundry page in [this PR](https://github.com/moonbeam-foundation/moonbeam-docs/pull/908/files#diff-486503116f9f0f2eaac893e5515b5aa270b5a948db271274e4f5604b645cc233)

It might be good to just do a quick scan over the page to make sure we didn't forget anything/in case anything else needs to be updated

File: builders/build/eth-api/dev-env/foundry.md [hunk 7]
```diff
@@ -191,7 +181,7 @@ forge script script/MyToken.s.sol --rpc-url https://rpc.api.moonbase.moonbeam.ne
 
 Foundry包括cast，一个用于执行以太坊RPC调用的CLI。
 
-尝试使用cast检索Token名称，其中`YOUR_CONTRACT_ADDRESS`是您在上一部分部署合约的地址：
+尝试使用Cast检索Token名称，其中`YOUR_CONTRACT_ADDRESS`是您在上一部分部署合约的地址：
```
Comment (eshaben) [inline builders/build/eth-api/dev-env/foundry.md:184]:
```suggestion
尝试使用Cast检索Token名称，其中`INSERT_YOUR_CONTRACT_ADDRESS`是您在上一部分部署合约的地址：
```

Comment (eshaben) [reply builders/build/eth-api/dev-env/foundry.md:184]: can you please update lines 189-207 to use `INSERT_YOUR_CONTRACT_ADDRESS`?

**Review by eshaben — COMMENTED** [review]
just a couple minor things! but looks good!

File: builders/build/eth-api/dev-env/foundry.md (hunk 7 above)
Comment (wuzhong-papermoon) [reply builders/build/eth-api/dev-env/foundry.md]: Done!

**Review by eshaben — APPROVED** [review]
LGTM!

**Pull request merged**